src/ewah.zig

```diff
@@ -61,98 +64,208 @@
             return @as(Word, @bitCast(mark));
         }
 
+        pub const Decoder = struct {
+            target_words: []Word,
+            target_index: usize = 0,
+            source_literal_words: usize = 0,
+
+            /// Returns the number of *words* written to `target_words` by this invocation.
+            // TODO Refactor to return an error when `source_chunk` is invalid,
+            // so that we can test invalid encodings.
+            pub fn decode(
+                decoder_: *Decoder,
```

Comment:
Let's try harder to avoid `thing_` names, as that goes against our intended style. Here, I thnik we
could have

```zig
Ewah.decode_all
Ewah.decode_chunks
```

to simultaneously free up decoder name, and also emphasize the connection between two decoding
methods.

And maybe call _this_ method `fn decode_chunk`?